src/test/java/com/artipie/rpm/RpmTest.java

```diff
@@ -89,4 +94,30 @@ void doesntBrakeMetadataWhenInvalidPackageSent(@TempDir final Path tmp)
             new IsEqual<>(repomd)
         );
     }
+
+    @Test
+    @Disabled
+    void showMeaningfulErrorWhenInvalidPackageSent() {
+        final Storage storage = new InMemoryStorage();
+        final Rpm repo =  new Rpm(
+            storage, StandardNamingPolicy.SHA1, Digest.SHA256, true
+        );
+        storage.save(
+            new Key.From("stored.rpm"),
+            new Content.From("stored content".getBytes())
+        );
+        repo.batchUpdate(Key.ROOT).blockingAwait();
+        final byte[] broken = {0x00, 0x01, 0x02 };
+        storage.save(
```

Comment:
@paulodamaso it's async call which returns `CompletableFuture`, you either need to wait it with `.toCompletableFuture().get()` or wrap this call with `BlockingStorage`